### PR TITLE
Reverse hints

### PR DESCRIPTION
**Pull request by terpstra**
Sometimes you might want to hint that dropping permissions is wise.

### CODE REVIEW

**Comment by terpstra** [comment]
Actually, I'm not sure it's a good idea to make this change until we have something that can actually issue these sorts of Hints.

File: src/main/scala/uncore/tilelink2/Bundles.scala [hunk 1]
```diff
@@ -104,10 +104,13 @@ object TLAtomics
...
 object TLHints
 {
-  val width = 1
+  val width = 3
 
-  def PREFETCH_READ  = UInt(0, width)
-  def PREFETCH_WRITE = UInt(1, width)
+  def GROW_toB = UInt(0, width)
+  def GROW_toT = UInt(1, width)
+  def SHRINK_toN = UInt(4, width)
```
Comment (hcook) [inline src/main/scala/uncore/tilelink2/Bundles.scala]: // flushes the block entirely

File: src/main/scala/uncore/tilelink2/Bundles.scala [hunk 2]
```diff
@@ -104,10 +104,13 @@ object TLAtomics
...
+  def SHRINK_toB = UInt(5, width)
```
Comment (hcook) [inline src/main/scala/uncore/tilelink2/Bundles.scala]: // produces the dirty data, while also ceding write permissions in anticipation of consumption

File: src/main/scala/uncore/tilelink2/Bundles.scala [hunk 3]
```diff
@@ -104,10 +104,13 @@ object TLAtomics
...
+  def SHRINK_toT = UInt(6, width)
```
Comment (hcook) [inline src/main/scala/uncore/tilelink2/Bundles.scala]: // cleans the block by writing back dirty data while retaining all permissions

File: src/main/scala/uncore/tilelink2/Bundles.scala [hunk 4]
```diff
@@ -104,10 +104,13 @@ object TLAtomics
 
 object TLHints
 {
-  val width = 1
+  val width = 3
 
-  def PREFETCH_READ  = UInt(0, width)
-  def PREFETCH_WRITE = UInt(1, width)
+  def GROW_toB = UInt(0, width)
```
Comment (hcook) [inline src/main/scala/uncore/tilelink2/Bundles.scala]: // acquires read permissions

File: src/main/scala/uncore/tilelink2/Bundles.scala [hunk 5]
```diff
@@ -104,10 +104,13 @@ object TLAtomics
...
+  def GROW_toT = UInt(1, width)
```
Comment (hcook) [inline src/main/scala/uncore/tilelink2/Bundles.scala]: // acquires read and write permissions

**Comment by terpstra** [comment]
I agree we should make this change, but the L2 change is non-trivial and I was hoping to have a test case before we go and break it without realizing.

**Comment by terpstra** [comment]
We've decided not to do this, and take a less "you must do this" approach.

**Comment by hcook** [comment]
Should cherry-pick 8c7f1c6

**Pull request closed without merging**